tempest-testing-internal/src/main/kotlin/app/cash/tempest/testing/internal/TestUtils.kt

```diff
@@ -28,37 +28,36 @@ import com.amazonaws.services.dynamodbv2.AmazonDynamoDBStreamsClientBuilder
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper
 import com.amazonaws.services.dynamodbv2.document.DynamoDB
 import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput
+import java.net.InetSocketAddress
 import java.net.ServerSocket
 import java.net.Socket
 
 fun pickRandomPort(): Int {
   ServerSocket(0).use { socket -> return socket.localPort }
 }
 
+private const val CONNECT_TIMEOUT_MILLIS = 1_000
+
 fun isServerListening(host: String, port: Int): Boolean {
-  var s: Socket? = null
-  return try {
-    s = Socket(host, port)
-    true
-  } catch (e: Exception) {
-    false
-  } finally {
-    if (s != null) try {
-      s.close()
+  return Socket().use {
+    try {
+      it.connect(InetSocketAddress(host, port), CONNECT_TIMEOUT_MILLIS)
+      true
     } catch (e: Exception) {
-      println("failed to close socket file")
+      false
     }
   }
 }
 
-fun connect(port: Int): AmazonDynamoDB {
```

Comment:
wild that a method named `connect` was previously making a socket connection as a side effect